rfq/negotiator_test.go

```diff
@@ -0,0 +1,349 @@
+package rfq
+
+import (
+	"testing"
+	"time"
+
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/fn"
+	"github.com/lightninglabs/taproot-assets/rfqmath"
+	"github.com/lightninglabs/taproot-assets/rfqmsg"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+// testCaseIncomingSellAccept is a test case for the handling of an incoming
+// sell accept message.
+type testCaseIncomingSellAccept struct {
+	// Name is the name of the test case.
+	Name string
+
+	// IncomingSellAcceptRate is the rate in the incoming sell accept
+	// message.
+	IncomingSellAcceptRate rfqmsg.AssetRate
+
+	// PriceOracleAskPrice is the rate returned by the price oracle.
+	PriceOracleAskPrice rfqmsg.AssetRate
+
+	// AcceptPriceDeviationPpm is the acceptable price deviation in ppm.
+	AcceptPriceDeviationPpm uint64
+
+	// QuoteRespStatus is the expected status of the quote check.
+	QuoteRespStatus fn.Option[QuoteRespStatus]
+}
+
+// assertIncomingSellAcceptTestCase asserts the handling of an incoming sell
+// accept message for a test case.
+func assertIncomingSellAcceptTestCase(
+	t *testing.T, tc testCaseIncomingSellAccept) {
+
+	// Create a mock price oracle.
+	mockPriceOracle := &MockPriceOracle{}
+
+	// Register an expected call and response for price oracle method
+	// QueryAskPrice.
+	mockPriceOracle.On(
+		"QueryAskPrice", mock.Anything, mock.Anything,
+		mock.Anything, mock.Anything, mock.Anything,
+	).Return(
+		&OracleResponse{
+			AssetRate: tc.PriceOracleAskPrice,
+		}, nil,
+	)
+
+	// Define sell request and sell accept messages.
+	var (
+		assetSpecifier = asset.NewSpecifierFromId(asset.ID{1, 2, 3})
+		peerID         = route.Vertex{1, 2, 3}
+		msgID          = rfqmsg.ID{1, 2, 3}
+	)
+
+	sellRequest := rfqmsg.SellRequest{
+		Peer:           peerID,
+		ID:             msgID,
+		AssetSpecifier: assetSpecifier,
+		PaymentMaxAmt:  lnwire.MilliSatoshi(1000),
+		AssetRateHint:  fn.None[rfqmsg.AssetRate](),
+	}
+
+	sellAccept := rfqmsg.SellAccept{
+		Request:   sellRequest,
+		AssetRate: tc.IncomingSellAcceptRate,
+	}
+
+	// Create the negotiator.
+	errChan := make(chan error, 1)
+	negotiator, err := NewNegotiator(NegotiatorCfg{
+		PriceOracle:             mockPriceOracle,
+		OutgoingMessages:        make(chan rfqmsg.OutgoingMsg, 1),
+		AcceptPriceDeviationPpm: tc.AcceptPriceDeviationPpm,
+		ErrChan:                 errChan,
+	})
+	require.NoError(t, err)
+
+	// Define the finalise callback function.
+	finalise := func(
```

Comment:
nit: format like other function definitions instead? Her and in other test below.

```
	finalise := func(msg rfqmsg.BuyAccept,
		event fn.Option[InvalidQuoteRespEvent]) {
```